dd-java-agent/instrumentation/jersey/src/main/java/datadog/trace/instrumentation/jersey/JerseyRequestExtractAdapter.java

```diff
@@ -0,0 +1,53 @@
+package datadog.trace.instrumentation.jersey;
+
+import io.opentracing.propagation.TextMap;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.core.MultivaluedMap;
+import org.glassfish.jersey.server.ContainerRequest;
+
+public class JerseyRequestExtractAdapter implements TextMap {
+  private final MultivaluedMap headers;
+
+  JerseyRequestExtractAdapter(final ContainerRequest request) {
+    headers = request.getHeaders();
+  }
+
+  @Override
+  public Iterator<Map.Entry<String, String>> iterator() {
+    HashMap<String, String> delinkedListHeaders = new HashMap<>();
+
+    // Jersey headers are Map<String, List<String>> so multiple values need to be flattened
+    for (final Object e : headers.entrySet()) {
+      Map.Entry<String, List<String>> entry = (Map.Entry<String, List<String>>) e;
+      final String key = entry.getKey().toLowerCase();
+      final String val = joinCsv(entry.getValue());
```

Comment:
I'm not sure joining multiple B3 propagation fields with commas is valid. Should this be looking for specific headers instead of flattening all?

Reply:
This is just to get the headers in accepted carrier format for the tracer's actual extract method: https://github.com/signalfx/signalfx-java-tracing/blob/master/dd-trace-ot/src/main/java/datadog/opentracing/propagation/B3HttpCodec.java#L71

This is similar to other instrumentations like servlet: https://github.com/signalfx/signalfx-java-tracing/blob/master/dd-java-agent/instrumentation/servlet-3/src/main/java/datadog/trace/instrumentation/servlet3/HttpServletRequestExtractAdapter.java#L75.

Given that a number of frameworks allow multiple header values, it's definitely important for us to ensure that during injection we use whatever single value setter is available to never allow joined B3 values: https://github.com/signalfx/signalfx-java-tracing/commit/feee7369450164a46dd30af726f5e106678891fe.

Reply:
On second thought it may make more sense just the use the first value instead of unnecessarily joining.  Nothing but the extraction is using this and it doesn't care about other headers.